tests/hscRepository.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python
+#
+# LSST Data Management System
+#
+# Copyright 2008-2015  AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+
+import os
+import unittest
+from glob import glob
+from shutil import rmtree
+from subprocess import check_call
+from tempfile import mkdtemp
+
+import lsst.afw.geom as afwGeom
+import lsst.daf.persistence as dafPersist
+import lsst.pex.exceptions as pexExcept
+import lsst.utils.tests as utilsTests
+from lsst.utils import getPackageDir
+
+TEST_DATA_PACKAGE = "testdata_subaru"
+
+def requirePackage(packageName):
+    """
+    Decorator to skip unit tests unless the package ``packageName`` is set up.
+    """
+    try:
+        getPackageDir(packageName)
+    except pexExcept.NotFoundError as e:
+        return unittest.skip("Required package %s not available [%s]." % (packageName, str(e)))
+    return lambda func: func
+
+
+def dataRepository(mapperName, inputPath):
+    """
+    Construct a data repository for a particular mapper containing a given set
+    of input data and return its path.
+
+    @param[in] mapperName   Name of mapper class for use with repository.
+    @param[in] inputPath    Location of data files which will be ingested.
+    """
+    repoPath = mkdtemp()
+    with open(os.path.join(repoPath, "_mapper"), "w") as _mapper:
+        _mapper.write(mapperName)
+    ingest_cmd = os.path.join(getPackageDir("obs_subaru"), "bin.src", "hscIngestImages.py")
+    check_call([ingest_cmd, repoPath] + glob(os.path.join(inputPath, "*.fits.gz")))
+    return repoPath
+
+
+class IngestRawTestCase(utilsTests.TestCase):
+    """Ensure that we can ingest raw data to a repository."""
+    @requirePackage(TEST_DATA_PACKAGE)
+    def setUp(self):
+        rawPath = os.path.join(getPackageDir(TEST_DATA_PACKAGE), "hsc", "raw")
+        self.repoPath = dataRepository("lsst.obs.hsc.HscMapper", rawPath)
+
+    def tearDown(self):
+        rmtree(self.repoPath)
+
+    def testIngest(self):
+        self.assertTrue(os.path.exists(os.path.join(self.repoPath, "registry.sqlite3")))
+
+    def testMapperName(self):
+        butler = dafPersist.Butler(root=self.repoPath)
+        self.assertEqual(butler.mapper.packageName, "obs_subaru")
+
```

Comment:
I think it is, because the different test cases test different things. It's useful to know if the various components of `IngestRawTestCase` are succeeding even if a problem means that `setUp()` for `GetDataTestCase` fails.

The duplication is minimal, so I'm minded to leave it as it is. If you strongly object, do say so.